### PR TITLE
[Improve][project-build] Limit the number of running build projects

### PR DESCRIPTION
**Pull request by zhilinli123**
## What changes were proposed in this pull request

Optimizing for multiple projects to run at the same time causes excessive cpu load on the machine

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by zhilinli123** [comment]
5f1b8b26-935e-43d6-9b80-256823247d88
363b6882-f8a9-4926-af65-35cccf6cad17

File: streampark-console/streampark-console-service/src/main/resources/config.yaml [hunk 1]
```diff
@@ -61,6 +61,9 @@ streampark:
     http-auth: 'simple'  # default simple, or kerberos
   # flink on yarn or spark on yarn, HADOOP_USER_NAME
   hadoop-user-name: hdfs
+  project:
+      # Number of projects allowed to be running at the same time , If there is no limit, -1 can be configured
+    build-num: 6
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/resources/config.yaml:66]: how  about max-build?

File: streampark-console/streampark-console-service/src/main/resources/mapper/core/ProjectMapper.xml [hunk 2]
```diff
@@ -84,5 +84,11 @@
             </if>
         </where>
     </select>
+    <select id="selectCurrentBuildCount" resultType="java.lang.Long">
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/resources/mapper/core/ProjectMapper.xml:87]:
```
<select id="getBuildingCount" resultType="java.lang.Long">
        select count(1)
        from t_flink_project
        where build_state = 0
  </select>
  ```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/ProjectService.java [hunk 3]
```diff
@@ -57,4 +57,11 @@ public interface ProjectService extends IService<Project> {
   List<Application> getApplications(Project project);
 
   boolean checkExists(Project project);
+
+  /**
+   * Gets current running build project data
+   *
+   * @return build number
+   */
+  Long selectCurrentBuildCount();
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/ProjectService.java:66]: How about change the method name to getBuildingCount?

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/ProjectController.java [hunk 4]
```diff
@@ -77,6 +82,16 @@ public RestResponse get(Project project) {
   @RequiresPermissions("project:build")
   @PermissionScope(team = "#project.teamId")
   public RestResponse build(Project project) throws Exception {
+    Long currentBuildCount = projectService.selectCurrentBuildCount();
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/ProjectController.java:85]: Don’t write logic code at the Controller layer, You can move the code in to the build method of ProjectServiceImpl, 

```
 ApiAlertException.throwIfTrue(
          currentBuildCount > maxProjectBuildNum,
          String.format(
              "The number of running Build projects exceeds the maximum number: %d of project max build",
              maxProjectBuildNum)
      );

```

**Pull request closed without merging**